ui/client/components/graph/node-modal/editors/expression/ExpressionField.js

```diff
@@ -0,0 +1,77 @@
+import React from "react"
+import _ from "lodash"
+import EditableExpression from "./EditableExpression"
+import ProcessUtils from "../../../../../common/ProcessUtils"
+import EditableExpressionWithTestResults from "./tests/EditableExpressionWithTestResults"
+import {Types} from "./EditorType"
+
+export default class ExpressionField extends React.Component {
+
+  render() {
+    const {
+      fieldName, fieldLabel, exprPath, validators, isEditMode, editedNode, isMarked, showValidation, showSwitch,
+      nodeObjectDetails, setNodeDataAt, testResultsToShow, testResultsToHide, toggleTestResult, renderFieldLabel, fieldType
+    } = this.props
+    const readOnly = !isEditMode
+    const exprTextPath = `${exprPath}.expression`
+    const expressionObj = _.get(editedNode, exprPath)
+    const marked = isMarked(exprTextPath)
+    const restriction = this.getRestriction(fieldName)
+
+    if (restriction.hasFixedValues)
+      return (
+        <EditableExpression
+          fieldType={"expressionWithFixedValues"}
+          fieldLabel={fieldLabel}
+          expressionObj={expressionObj}
+          renderFieldLabel={renderFieldLabel}
+          values={restriction.values}
+          isMarked={marked}
+          showSwitch={showSwitch}
+          readOnly={readOnly}
+          onValueChange={(newValue) => setNodeDataAt(exprTextPath, newValue)}
+        />
+      )
+
+    const param = this.findParamByName(fieldLabel)
+
+    const editableExpression =
+      <EditableExpression
+        fieldType={fieldType || (param ? ProcessUtils.humanReadableType(param.typ.refClazzName) : "expression")}
```

Comment:
This shoud be done in EditableExpression.

Reply:
Done